packages/kbn-expandable-flyout/src/utils/helpers.test.ts

```diff
@@ -0,0 +1,526 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { FlyoutState, initialFlyoutState } from '../models/state';
+import {
+  closeFlyout,
+  closeFlyoutLeftPanel,
+  closeFlyoutPreviewPanel,
+  closeFlyoutRightPanel,
+  openFlyout,
+  openFlyoutLeftPanel,
+  openFlyoutPreviewPanel,
+  openFlyoutRightPanel,
+  previousFlyoutPreviewPanel,
+  selectFlyoutLayout,
+} from './helpers';
+import { FlyoutLayout, FlyoutPanel } from '../..';
+
+const rightPanel1: FlyoutPanel = {
+  id: 'right1',
+  path: ['path'],
+};
+const leftPanel1: FlyoutPanel = {
+  id: 'left1',
+  params: { id: 'id' },
+};
+const previewPanel1: FlyoutPanel = {
+  id: 'preview1',
+  state: { id: 'state' },
+};
+
+const rightPanel2: FlyoutPanel = {
+  id: 'right2',
+  path: ['path'],
+};
+const leftPanel2: FlyoutPanel = {
+  id: 'left2',
+  params: { id: 'id' },
+};
+const previewPanel2: FlyoutPanel = {
+  id: 'preview2',
+  state: { id: 'state' },
+};
+
+describe('helpers', () => {
+  describe('openFlyout', () => {
+    it('should add panels to empty state', () => {
+      const initialState: FlyoutState = initialFlyoutState;
```

Comment:
nit: check out this, it helps with repeated logic like the one you have here. some people dont like it though https://jestjs.io/docs/api#describeeachtablename-fn-timeout , I don't blame 'em

Reply:
that's pretty cool, but here also I kinda like having it in the test directly, makes it easy to read (to my eyes)